### DISCUSSION TITLE
UseFieldArray appending empty objects. (ESLint)

### DISCUSSION THREAD

**Discussion by Zanderinos** [react-hook-form/react-hook-form#8892] [Q&A]
**Describe the bug/issue**
The data for appending data is now required and not partial since we tried to upgrade "react-hook-form from 7.33.0 to 7.34.2" This is causing our forms to break because we sometimes only append empty objects for creating empty input fields. A quick solution would be to add empty strings as default values, but this will not always work in our case.

**To Reproduce**
1. Create a simple yup validation schema
` yup.array(
      yup.object({
         firstName: yup.string().required(),
         lastName: yup.string().required(),
        })
        .required(),
    )`
2. Create a button to call `append` on the `useFieldArray `
`userFieldArray.append({ firstName: undefined, lastName: undefined })`
3. Eslint will display the following error:

![Screenshot 2022-08-22 at 12 26 57](https://user-images.githubusercontent.com/5887292/185900111-6a926439-5d89-4740-974f-148b72bd00b9.png)

**Question**
I was wondering if my implementation is wrong or is there some kind of workaround to append without adding data?

**Comment by Moshyfawn** [comment]
Check out [the rules section of RHF docs on `useForm` props](https://react-hook-form.com/api/useform#props):

> **Important:** You should provide a proper default value and avoid undefined.
> - `undefined` is reserved for fallback from inline `defaultValue`/`defaultChecked` to hook level `defaultValues`.
> - `undefined` value is conflicting with controlled component as default state

**Reply by Moshyfawn** [reply]
Additionally, the `useFieldArray` rules section reference:

> When you `append`, `prepend`, `insert` and `update` the field array, the obj can't be empty object rather need to supply all your input's `defaultValues`.

> ```js
> append(); ❌
> append({}); ❌
> append({ firstName: 'bill', lastName: 'luo' }); ✅
> ```

**Reply by Zanderinos** [reply]
> Additionally, the `useFieldArray` rules section reference:
> 
> > When you `append`, `prepend`, `insert` and `update` the field array, the obj can't be empty object rather need to supply all your input's `defaultValues`.
> 
> > ```js
> > append(); ❌
> > append({}); ❌
> > append({ firstName: 'bill', lastName: 'luo' }); ✅
> > ```
Thanks for the explanation. 

What if I have a form asking someone how many children they have, the expected type would be a number. What will be the default value then? Because if I default to 0 it will not error on submit, but 0 isn't necessarily the correct answer. So I rather want it to be an empty value so it will error on submit and requesting them to explicitly provide the amount of children to prevent the user from submitting the wrong value accidentally.

**Reply by Moshyfawn** [reply]
`null` is another valid empty value

**Reply by dborysov** [reply]
Hi, @Moshyfawn! 

Could you help with the [following example](https://stackblitz.com/edit/react-ts-qedcvb?file=App.tsx)? It's really similar to what @Zanderinos mentioned, but maybe it would be more clear to us with the stackblitz example.

For some reason Stackblitz doesn't show the error from the beginning, so you need to trigger Save in tsconfig.json file
